packages/devtools/lib/src/main.dart

```diff
@@ -128,8 +130,75 @@ class PerfToolFramework extends Framework {
     showError(title, error);
   }
 
-  void initTestingModel() {
-    App.register(this);
+  Future<bool> isFlutterApp() async {
+    final EvalOnDartLibrary flutterLibrary = EvalOnDartLibrary(
+      flutterLibraryUriCandidates,
+      serviceManager.service,
+    );
+
+    try {
+      await flutterLibrary.libraryRef.catchError(
+        (_) => throw FlutterLibraryNotFound(),
+        test: (e) => e is LibraryNotFound,
+      );
+    } catch (e) {
+      if (e is FlutterLibraryNotFound) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  Future<bool> isProfileBuild() async {
+    try {
+      final Isolate isolate = await serviceManager.service
+          .getIsolate(serviceManager.isolateManager.selectedIsolate.id);
+      await serviceManager.service.evaluate(
+        serviceManager.isolateManager.selectedIsolate.id,
+        isolate.rootLib.id,
+        '1+1',
+      );
+      // If we reach this return statement, no error was thrown and this is not
+      // a profile build.
+      return false;
+    } catch (e) {
+      if (e is RPCError) {
```

Comment:
you can catch only specific error types in a catch block. Use that instead of an is check here and elsewhere.